new/language/implementation/implementation.go

```diff
@@ -107,6 +107,18 @@ type Implementation interface {
 	// we want to ignore member_expressions as roots.
 	IsRootOfRuleQuery(node *tree.Node) bool
 	PatternLeafContentTypes() []string
+	// ShouldSkipNode returns wether a node should should be skipped or assigned variable to it
+	// it is useful for cases when you have nested nodes to ignore and want to assign variable to its child
+	//
+	// eg. given the following tree sitter
+	// arguments
+	//	formal_parameters
+	//		required_parameter
+	//			identifier
+	//
+	//  if you want to get only identifier instead of required parameter ShouldSkipNode should return true
+	//  for required parameter
+	ShouldSkipNode(node *tree.Node) bool
```

Comment:
I think this would be better inverted. ie. `IsPatternVariableNodeType`.

For JS/TS we should just need `identifier` and for Ruby `identifier` and `constant`.

Reply:
I've tried that and got multiple ruby and javascript rules failures.
I find it simpler to define a single node type we want to skip than to define 3+ node types we want to parse as variables.